### PR TITLE
fix failing tests and warnings

### PR DESCRIPTION
**Pull request by JuliaSprenger**

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Is it still a draft or can we merge ?

**Comment by JuliaSprenger** [comment]
@samuelgarcia Now it's ready for review.
This reduces the warnings from ~3000 to only a few that are triggered by imported packages on python 3.9

File: neo/test/iotest/test_nixio.py [hunk 1]
```diff
@@ -1526,7 +1527,7 @@ def test_annotation_types(self):
             "sometime": time(13, 37, 42),
             "somequantity": self.rquant(10, pq.ms),
             "somestring": self.rsentence(3),
-            "npfloat": np.float(10),
+            "npfloat": float(10),
```
Comment (samuelgarcia) [inline neo/test/iotest/test_nixio.py:1530]: here I think we are testing specific type. I would go for np.float64

**Comment by samuelgarcia** [comment]
Hi Julia. 
C'est parfait! Merci.
I made a very small comment.

File: neo/test/iotest/test_nixio.py (hunk 1 above)
Comment (JuliaSprenger) [reply neo/test/iotest/test_nixio.py:1530]:
```suggestion
            "npfloat": np.float64(10),
```

**Comment by samuelgarcia** [comment]
Cool.
Can I merge?

**Comment by JuliaSprenger** [comment]
Jupp, go ahead!

**Pull request merged**